System.Win32 exports `x`

It seems like `System.Win32` now exports a few rather common symbols, including `x`. It will cause quite some inconvenience for such a widely-used module to export such a common name. Was this intentional?

@Mistuke, for instance, Win32's own `registry001` test breaks due to this:
```
[1 of 1] Compiling Main             ( registry001.hs, registry001.o )

registry001.hs:14:29: error:
    Ambiguous occurrence ‘x’
    It could refer to
       either ‘System.Win32.x’,
              imported from ‘System.Win32’ at registry001.hs:1:1-19
              (and originally defined in ‘System.Win32.Console’)
           or ‘Main.x’, defined at registry001.hs:5:1
```

 it seems that this is due to the `x` and `y` fields of the `System.Win32.Console.COORD` type, which is reexported from `System.Win32`. The `left` and `right` fields of `SMALL_RECT` will be similarly problematic.

--- a/System/Win32/Console.hsc
+++ b/System/Win32/Console.hsc
@@ -170,8 +170,8 @@ instance Storable CONSOLE_SCREEN_BUFFER_INFO where
         (#poke CONSOLE_SCREEN_BUFFER_INFO, dwMaximumWindowSize) buf (dwMaximumWindowSize info)
 
 data COORD = COORD
-    { x :: SHORT
-    , y :: SHORT
+    { xPos :: SHORT
+    , yPos :: SHORT
     } deriving (Show, Eq)
 
 instance Storable COORD where
@@ -182,14 +182,14 @@ instance Storable COORD where
         y' <- (#peek COORD, Y) buf
         return $ COORD x' y'
     poke buf coord = do
-        (#poke COORD, X) buf (x coord)
-        (#poke COORD, Y) buf (y coord)
+        (#poke COORD, X) buf (xPos coord)
+        (#poke COORD, Y) buf (yPos coord)
 
 data SMALL_RECT = SMALL_RECT
-    { left   :: SHORT
-    , top    :: SHORT
-    , right  :: SHORT
-    , bottom :: SHORT
+    { leftPos   :: SHORT
+    , topPos    :: SHORT
+    , rightPos  :: SHORT
+    , bottomPos :: SHORT
     } deriving (Show, Eq)
 
 instance Storable SMALL_RECT where
@@ -202,10 +202,10 @@ instance Storable SMALL_RECT where
         bottom' <- (#peek SMALL_RECT, Bottom) buf
         return $ SMALL_RECT left' top' right' bottom'
     poke buf small_rect = do
-        (#poke SMALL_RECT, Left) buf (left small_rect)
-        (#poke SMALL_RECT, Top) buf (top small_rect)
-        (#poke SMALL_RECT, Right) buf (right small_rect)
-        (#poke SMALL_RECT, Bottom) buf (bottom small_rect)
+        (#poke SMALL_RECT, Left) buf (leftPos small_rect)
+        (#poke SMALL_RECT, Top) buf (topPos small_rect)
+        (#poke SMALL_RECT, Right) buf (rightPos small_rect)
+        (#poke SMALL_RECT, Bottom) buf (bottomPos small_rect)
 
 foreign import WINDOWS_CCONV safe "windows.h GetConsoleScreenBufferInfo"
     c_GetConsoleScreenBufferInfo :: HANDLE -> Ptr CONSOLE_SCREEN_BUFFER_INFO -> IO BOOL

--- a/System/Win32/Time.hsc
+++ b/System/Win32/Time.hsc
@@ -33,7 +33,7 @@ import Foreign          ( Storable(sizeOf, alignment, peekByteOff, peek,
                         , with, alloca, allocaBytes, copyArray )
 import Foreign.C        ( CInt(..), CWchar(..)
                         , peekCWString, withCWStringLen, withCWString )
-import Foreign.Marshal.Utils (with, maybeWith)
+import Foreign.Marshal.Utils (maybeWith)
 
 ##include "windows_cconv.h"
 #include <windows.h>

--- a/changelog.md
+++ b/changelog.md
@@ -1,6 +1,6 @@
 # Changelog for [`Win32` package](http://hackage.haskell.org/package/Win32)
 
-## 2.9.1.0 September 2020
+## 2.10.0.0 September 2020
 
 * Add function `isWindowVisible`
 * Add function `getLastInputInfo`
@@ -14,6 +14,7 @@
 * Add `fromDateFormatPciture` and `fromTimeFormatPicture`, to translate from
   Windows date and time format pictures to format strings used by the `time`
   package.
+* Renamed fields of `COORD` and `SMALL_RECT` to avoid name clashes. (See #157)
 
 ## 2.9.0.0 June 2020
 

